packages/playground/remote/src/lib/worker-utils.ts

```diff
@@ -30,20 +31,28 @@ export type ParsedStartupOptions = {
 	storage: string;
 	phpExtensions: string[];
 	siteSlug: string;
+	scope: string;
 };
 
-export const receivedParams: ReceivedStartupOptions = {};
-const url = self?.location?.href;
-if (typeof url !== 'undefined') {
-	const params = new URL(self.location.href).searchParams;
-	receivedParams.wpVersion = params.get('wpVersion') || undefined;
-	receivedParams.phpVersion = params.get('phpVersion') || undefined;
-	receivedParams.storage = params.get('storage') || undefined;
-	// Default to CLI to support the WP-CLI Blueprint step
-	receivedParams.sapiName = params.get('sapiName') || 'cli';
-	receivedParams.phpExtensions = params.getAll('php-extension');
-	receivedParams.siteSlug = params.get('site-slug') || undefined;
-}
+const getReceivedParams = async () => {
+	return new Promise<ReceivedStartupOptions>((resolve) => {
+		self.addEventListener('message', async (event) => {
+			if (event.data.type === 'startup-options') {
+				resolve({
+					wpVersion: event.data.startupOptions.wpVersion,
+					phpVersion: event.data.startupOptions.phpVersion,
+					storage: event.data.startupOptions.storage,
+					sapiName: event.data.startupOptions.sapiName,
+					phpExtensions: event.data.startupOptions['php-extension'],
+					siteSlug: event.data.startupOptions['site-slug'],
+					scope: event.data.startupOptions.scope,
+				});
+			}
+		});
+	});
+};
+
+const receivedParams: ReceivedStartupOptions = await getReceivedParams();
```

Comment:
This await is a part of the boot flow but now it's concealed in the `worker-utils.ts` package. I understand the rationale – we need these params to continue initializing startupOptions. My worry is someone will one day spend hours debugging a boot issue without realizing the utils module also performs parts of the boot flow. What would be the easiest way of moving this await to `worker-thread.ts`?

Reply:
I agree, it's not expected that utils control the flow. [c85a682](https://github.com/WordPress/wordpress-playground/pull/1606/commits/c85a682e9d27d3828e3704c6062faa541ad71301) moves the await to the top of worker-thread but keeps everything else in worker-utils.

I considered moving `startupOptions` to the worker-thread, but that would require us to update `createPhpRuntime`, so it doesn't rely on startupOptions, or also move it to the worker-thread.
I didn't do this because I'm not sure if it's worth the effort.